tap_twitter_ads/streams.py

```diff
@@ -318,82 +326,82 @@ def sync_endpoint(self,
+                    # Loop thru cursor records, break out if no more data or bookmark_value < last_dttm
+                    for record in cursor:
+                        # Get dictionary for record
+                        record_dict = self.obj_to_dict(record)
+                        if not record_dict:
+                            # Finish looping
+                            LOGGER.info('Stream: {} - Finished Looping, no more data'.format(stream_name))
+                            break
+
+                        # Get record's bookmark_value
+                        # All bookmarked requests are sorted by updated_at descending
+                        #   'sort_by': ['updated_at-desc']
+                        # The first record is the max_bookmark_value
+                        if bookmark_field:
+                            bookmark_value_str = record_dict.get(bookmark_field)
+                            if bookmark_value_str:
+                                # Tweets use a different datetime format: '%a %b %d %H:%M:%S %z %Y'
+                                if datetime_format:
+                                    bookmark_value = datetime.strptime(
+                                        record_dict.get(bookmark_field), datetime_format)
+                                # Other bookmarked endpoints use normal UTC format
+                                else:
+                                    bookmark_value = strptime_to_utc(record_dict.get(bookmark_field))
+                                # If first record, set max_bookmark_value
+                                if i == 0:
+                                    max_bookmark_dttm = bookmark_value
+                                    max_bookmark_value = max_bookmark_dttm.strftime('%Y-%m-%dT%H:%M:%S%z')
+                                    LOGGER.info('Stream: {} - max_bookmark_value: {}'.format(
+                                        stream_name, max_bookmark_value))
                             else:
-                                bookmark_value = strptime_to_utc(record_dict.get(bookmark_field))
-                            # If first record, set max_bookmark_value
-                            if i == 0:
-                                max_bookmark_dttm = bookmark_value
-                                max_bookmark_value = max_bookmark_dttm.strftime('%Y-%m-%dT%H:%M:%S%z')
-                                LOGGER.info('Stream: {} - max_bookmark_value: {}'.format(
-                                    stream_name, max_bookmark_value))
+                                # pylint: disable=line-too-long
+                                LOGGER.info('Stream: {} - NO BOOKMARK, bookmark_field: {}, record: {}'.format(
+                                    stream_name, bookmark_field, record_dict))
+                                # pylint: enable=line-too-long
+                                bookmark_value = last_dttm
+                            if bookmark_value < last_dttm:
```

Comment:
Add a comment to explain why we are breaking the loop.

Reply:
Added comment.